test/imath/ref/out.xml

```diff
@@ -0,0 +1,517 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<TranslationUnit name="/home/anders/code/cppmm/test/imath/bind/imath_box.cpp" id="5">
+    <Record name="Imath_3_0::Box&lt;Imath_3_0::Vec3&lt;float&gt; &gt;" id="6" width="192" align="32">
+        <Attribute>cppmm:valuetype</Attribute>
+        <Field name="min"><QType const="0"><RecordType name="TYPE:Imath_3_0::Vec3&lt;float&gt;" id="1" type="Imath_3_0::Vec3&lt;float&gt;" record="27" />
+</QType></Field>
+        <Field name="max"><QType const="0"><RecordType name="TYPE:Imath_3_0::Vec3&lt;float&gt;" id="1" type="Imath_3_0::Vec3&lt;float&gt;" record="27" />
+</QType></Field>
+        <Method name="Imath_3_0::Box&lt;Imath_3_0::Vec3&lt;float&gt; &gt;::Box" id="7" short_name="Box" in_binding="0" in_library="0" is_static="0">
+            <Return><QType const="0"><BuiltinType name="TYPE:void" id="0" type="void" /></QType></Return>
+        </Method>
+        <Method name="Imath_3_0::Box&lt;Imath_3_0::Vec3&lt;float&gt; &gt;::Box" id="7" short_name="Box" in_binding="0" in_library="0" is_static="0">
+            <Return><QType const="0"><BuiltinType name="TYPE:void" id="0" type="void" /></QType></Return>
```

Comment:
what does const="0" mean for QType ? I mean if it's non const can it just be the BuiltinType directly ?

Reply:
Yes QType is basically QualType from clang (which also handles volatile and restrict but I don't think we care about those). Omitting the node when it's not const would look neater, but would complicate the code a lot (on the reading side as well I think?) so I'd rather leave it there.

Reply:
okay np

Reply:
I’ll reassess this assumption when I do the json version.

Reply:
was thinking this might be useful...
https://uscilab.github.io/cereal/

closest thing c++ has to serde..

Reply:
k, ignore the above comment about cereal. I've just tested it and it's using rtti.

Reply:
Using nlohmann::json now and it’s very simple and clean.

Reply:
I’ve also flattened qtype into the type in the output which works better. I also need to look at removing the NodeType nodes or keeping them in a separate tree so Type references can just be an id.